client/src/main/java/io/pravega/client/stream/impl/PositionImpl.java

```diff
@@ -106,7 +106,7 @@ public String toString() {
         return ToStringUtils.mapToString(ownedSegments);
     }
 
-    private static class PositionBuilder implements ObjectBuilder<PositionImpl> {
```

Comment:
Can we add a constructor of this `PositionImpl.java` that does not execute eagerly a `for` inside the constructor? Can this `for` be executed just if the user uses this object?